src/Tasks/AssemblyDependency/ResolveAssemblyReference.cs

```diff
@@ -1024,8 +1024,8 @@ List<Exception> generalResolutionExceptions
                                 {
                                     { "logMessage", output },
                                     { "logMessageDetails", details },
-                                    { "victorVersionNumber", victor.ReferenceVersion.ToString() },
-                                    { "victimVersionNumber", conflictCandidate.ReferenceVersion.ToString() }
+                                    { "victorVersionNumber", victor.ReferenceVersion?.ToString() },
```

Comment:
Would it be better to pass an empty string rather than null if `ReferenceVersion` is null?

Reply:
It doesn't seem to effect the outcome, does it? Either way there isn't a value for the metadata on this item.

Reply:
I wasn't sure how it would ultimately was used, so I looked into it. Looks like it null checks anyway. LGTM

```c#
            /// <summary>
            /// Returns the escaped value of the requested metadata name.
            /// </summary>
            string ITaskItem2.GetMetadataValueEscaped(string metadataName)
            {
                ErrorUtilities.VerifyThrowArgumentNull(metadataName, nameof(metadataName));

                string metadataValue = null;

                if (FileUtilities.ItemSpecModifiers.IsDerivableItemSpecModifier(metadataName))
                {
                    // FileUtilities.GetItemSpecModifier is expecting escaped data, which we assume we already are.
                    // Passing in a null for currentDirectory indicates we are already in the correct current directory
                    metadataValue = FileUtilities.ItemSpecModifiers.GetItemSpecModifier(null, _escapedItemSpec, _escapedDefiningProject, metadataName, ref _fullPath);
                }
                else if (_customEscapedMetadata != null)
                {
                    _customEscapedMetadata.TryGetValue(metadataName, out metadataValue);
                }

                return metadataValue ?? String.Empty;
            }
```

Reply:
I'm confused—what does this code have to do with this PR?